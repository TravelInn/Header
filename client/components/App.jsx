import React from "react";
import ReactDOM from ""

class App extends React.Component {
	constructor(props) {
		super(props);
		this.state ={

		}
	}

	render() {
		return(
			<p>hi from App</p>
		)
	}

}