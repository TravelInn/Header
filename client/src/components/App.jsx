import React from "react";
import ReactDOM from "react-dom";
import Features from "./Features.jsx";
import Icons from "./Icons.jsx";

class App extends React.Component {
	constructor(props) {
		super(props);
		this.state = {};
	}

	render() {
		return (
			<div>
				<Icons />
				<Features />
				<div id="hostelName"> ISLAND HOSTELS </div>
				<div id="location">
					<i className="fas fa-map-marker-alt" />
					48 Station Road, Mountain Lavinia, Colombo, Sri Lanka
				</div>
				<div className="modal fade" id="exampleModal" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
				  <div className="modal-dialog" role="document">
				    <div className="modal-content">
				      <div className="modal-header">
				        <button type="button" className="close" data-dismiss="modal" aria-label="Close">
				          <span aria-hidden="true">&times;</span>
				        </button>
				        <div className="modal-title" id="exampleModalLabel">Sign in or <a
				        href="https://tsecure.hostelworld.com/en/signup">Sign up</a>
				        </div>
				      </div>
				      <div className="modal-body">
				        <form role="form">
		                    <div class="form-group">
		                      <label for="usrname"> EMAIL ADDRESS/ USERNAME</label>
		                      <input type="text" class="form-control" id="usrname" placeholder="Username/email address" />
		                    </div>
		                    <div class="form-group">
		                      <label for="psw">PASSWORD</label>
		                      <input type="text" class="form-control" id="psw" placeholder="Password"/>
		                    </div>
				        </form>
				      </div>
				      <div className="modal-footer">
				      	<a> Forgot password? </a>
				      	<button type="button" className="btn btn-search">Login</button>
				      </div>
				    </div>
				  </div>
				</div>
			</div>
		);
	}
}

export default App;
