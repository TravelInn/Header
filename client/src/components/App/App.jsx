import React from "react";
import ReactDOM from "react-dom";
import axios from "axios";
import Wifi from "../Wifi.jsx";
import Modal from "../Modal.jsx";
import Coffee from "../Coffee.jsx";
import Icons from "../Icons.jsx";
import Carousel from "../Carousel.jsx";

class App extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			carousel: false,
			search: false,
			wifi: false,
			coffee: false,
			name: "",
			location: "",
			city: "",
			country: "",
			photos: [],
			id: ""
		};
		this.getHostelInfo = this.getHostelInfo.bind(this);
		this.getLocationInfo = this.getLocationInfo.bind(this);
		this.toggleCarousel = this.toggleCarousel.bind(this);
		this.openSearch = this.openSearch.bind(this);
	}

	componentDidMount() {
		this.getHostelInfo();
		this.getLocationInfo();
	}

	getHostelInfo() {
		axios
			.get(`/api/locations/hostels/99-178-4713/info`)
			.then(response => {
				const features = response.data[0].features[0];
				if (features.wifi) {
					this.setState({
						wifi: true
					});
				}

				if (features.coffee) {
					this.setState({
						coffee: true
					});
				}
				this.setState({
					name: response.data[0].hostel_name,
					location: response.data[0].street_name,
					photos: response.data[0].photos
				});
			})
			.catch(error => {
				console.log(error);
			});
	}

	getLocationInfo() {
		axios.get("/api/locations/99/info").then(response => {
			this.setState({
				city: response.data[0].city,
				country: response.data[0].country
			});
		});
	}

	toggleCarousel() {
		if (this.state.search === false) {
			this.setState(prevState => ({
				carousel: !prevState.carousel
			}));
		}
	}

	openSearch() {
		this.setState(prevState => ({
			search: !prevState.search
		}));
	}

	render() {
		return (
			<div>
				<Icons
					languages={this.props.languages}
					currency={this.props.currency}
					guests={this.props.guests}
					search={this.state.search}
					openSearch={this.openSearch}
				/>
				<div
					onClick={() => {
						this.toggleCarousel();
					}}
					id="showCarousel1"
				/>
				<div
					onClick={() => {
						this.toggleCarousel();
					}}
					id="showCarousel2"
				/>
				{this.state.wifi ? <Wifi /> : null}
				{this.state.coffee ? <Coffee /> : null}
				<div id="breadcrumbs">
					{" "}
					Home / {this.state.country} / {this.state.city}
				</div>
				<div id="hostelName"> {this.state.name} </div>
				<div id="location">
					<i className="fas fa-map-marker-alt" />
					{this.state.location}, {this.state.city},{" "}
					{this.state.country}
				</div>
				<Modal />
				{this.state.carousel ? (
					<Carousel
						photos={this.state.photos}
						closeCarousel={this.toggleCarousel}
					/>
				) : null}
			</div>
		);
	}
}

export default App;
