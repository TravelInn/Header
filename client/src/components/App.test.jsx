import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

import React from "react";
import { shallow } from "enzyme";
// import { shallowToJSON } from "enzyme-to-json";

// https://github.com/vjwilson/enzyme-example-jest

import App from "./App";

describe("App", () => {
	test("should render correctly", () => {
		expect(shallow(<App />)).toMatchSnapshot();
	});
});
