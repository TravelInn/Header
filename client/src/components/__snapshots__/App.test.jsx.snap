// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icons
          toggleCarousel={[Function]}
        />,
        <div
          id="showCarousel1"
          onClick={[Function]}
        />,
        null,
        null,
        <div
          id="breadcrumbs"
        >
           
          Home / 
          
           / 
          
        </div>,
        <div
          id="hostelName"
        >
           
          
           
        </div>,
        <div
          id="location"
        >
          <i
            className="fas fa-map-marker-alt"
          />
          
          , 
          
          ,
           
          
        </div>,
        <Modal />,
        null,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "currency": undefined,
          "guests": undefined,
          "languages": undefined,
          "toggleCarousel": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "showCarousel1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      null,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "Home / ",
            "",
            " / ",
            "",
          ],
          "id": "breadcrumbs",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Home / ",
          "",
          " / ",
          "",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "",
            " ",
          ],
          "id": "hostelName",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "",
          " ",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fas fa-map-marker-alt"
            />,
            "",
            ", ",
            "",
            ",",
            " ",
            "",
          ],
          "id": "location",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-map-marker-alt",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "",
          ", ",
          "",
          ",",
          " ",
          "",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icons
            toggleCarousel={[Function]}
          />,
          <div
            id="showCarousel1"
            onClick={[Function]}
          />,
          null,
          null,
          <div
            id="breadcrumbs"
          >
             
            Home / 
            
             / 
            
          </div>,
          <div
            id="hostelName"
          >
             
            
             
          </div>,
          <div
            id="location"
          >
            <i
              className="fas fa-map-marker-alt"
            />
            
            , 
            
            ,
             
            
          </div>,
          <Modal />,
          null,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currency": undefined,
            "guests": undefined,
            "languages": undefined,
            "toggleCarousel": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "showCarousel1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        null,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "Home / ",
              "",
              " / ",
              "",
            ],
            "id": "breadcrumbs",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Home / ",
            "",
            " / ",
            "",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "",
              " ",
            ],
            "id": "hostelName",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "",
            " ",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-map-marker-alt"
              />,
              "",
              ", ",
              "",
              ",",
              " ",
              "",
            ],
            "id": "location",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-map-marker-alt",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "",
            ", ",
            "",
            ",",
            " ",
            "",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
